src/wallet/wallet2.cpp

```diff
@@ -370,23 +370,19 @@ std::unique_ptr<tools::wallet2> make_basic(const boost::program_options::variabl
   if (command_line::is_arg_defaulted(vm, opts.daemon_host) && command_line::is_arg_defaulted(vm, opts.daemon_port) && command_line::is_arg_defaulted(vm, opts.daemon_address))
     daemon_address = tools::wallet2::get_default_daemon_address();
 
-  std::string default_protocol = "http://";
-  // Deprecated --daemon-ssl option: prepend https:// if there is no protocol on the daemon address
-  if (command_line::get_arg(vm, opts.daemon_ssl) == "enabled") {
-    default_protocol = "https://";
-    THROW_WALLET_EXCEPTION_IF(tools::starts_with(daemon_address, "http://"), tools::error::wallet_internal_error,
-        "Deprecated --daemon-ssl=enabled option conflicts with http://... daemon URL");
-  }
-
   if (daemon_address.empty())
   {
     daemon_address = (daemon_host.empty() ? "localhost" : daemon_host) + ':' +
       std::to_string(daemon_port > 0 ? daemon_port : get_config(nettype).RPC_DEFAULT_PORT);
   }
 
-  if (!std::regex_search(daemon_address, protocol_re))
-  {
-    daemon_address.insert(0, default_protocol);
+  std::string default_protocol = "http://";
```

Comment:
Unused but I suspect we want it to be used as we don't prepend a protocol like we do in the updated `set_daemon` that we do now.

Reply:
Oops, no, that was supposed to be deleted.

Reply:
We want to make HOSTNAME turn into http://HOSTNAME ?

Reply:
Yes, but it isn't necessary to do it here; that `http://` gets added already in wallet2's `set_daemon` if it isn't present, which this eventually gets to (and is also called from other places).

Reply:
Ok LGTM